### PR TITLE
Allow empty urdf tag

### PR DESCRIPTION
**Pull request by fmauch**
This is #1008 with updated tests.

### CODE REVIEW

File: hardware_interface/test/test_component_parser.cpp [hunk 1]
```diff
@@ -100,14 +100,14 @@ TEST_F(TestComponentParser, component_interface_type_empty_throws_error)
...
-TEST_F(TestComponentParser, parameter_empty_throws_error)
+TEST_F(TestComponentParser, parameter_empty_throws_no_error)
 {
   const std::string broken_urdf_string =
     std::string(ros2_control_test_assets::urdf_head) +
     ros2_control_test_assets::invalid_urdf_ros2_control_parameter_empty +
     ros2_control_test_assets::urdf_tail;
 
-  ASSERT_THROW(parse_control_resources_from_urdf(broken_urdf_string), std::runtime_error);
+  ASSERT_NO_THROW(parse_control_resources_from_urdf(broken_urdf_string));
```
Comment (bmagyar) [inline hardware_interface/test/test_component_parser.cpp:110]: While this may be technically correct. I don't think we should do it. We need a new test explicitly to make sure that empty tags as here are permitted but not a fully broken urdf as per the tests suggests

Comment (fmauch) [reply hardware_interface/test/test_component_parser.cpp]: Thanks for the feedback, absolutely valid point. I replaced the test with a more consistent one in 4c02c65.

**Review by destogl — APPROVED** [review]
Thanks!!

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
@mergifyio backport humble
